zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileInfo.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.notebook;
+
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Represents note file info for storage 
+ *
+ */
+public class FileInfo {
+
+  private ConcurrentMap<String, String> fileName;
+  private boolean forceRename;
+  public static final FileInfo EMPTY = new FileInfo();
+  
+  private FileInfo() {
+    fileName = Maps.newConcurrentMap();
+    forceRename = false;
+  }
+  
+  public static FileInfo createInstance() {
+    return new FileInfo();
+  }
+  
+  public String getFileName(String storageName) {
+    if (!fileName.containsKey(storageName)) {
```

Comment:
Note will trigger that operation when each time note title is changed. IMHO, where to store the note and in what kind of format should be transparent to Note. Note just need to call `NotebookRepo` to save or update via its interface. It is `NotebookRepo`'s responsibility to map note to some object in the storage system (file for `VFSNotebookRepo`). 

Another thing I am thinking is that whether the rename operation should just happen in `getNote` instead of `saveNote`.  So basically there would be 3 apis in `NotebookRepo`.  
 
* getNote
* renameNote
* saveNote

The renaming operation could happen in `getNote` & `renameNote`. The renaming in `getNote` is for old/new version conversion, the `renameNote` is for title update (Actually it depends on the implementation, some `NotebookRepo` may not use title as file name or they may not use file to store, such as hbase)


My point is that we should make the boundary between `Note` and `NotebookRepo` clean. The interaction between these 2 things only through interfaces. Anyway this may involve some refactor, may not be related to this PR. But it would benefit zeppelin for long term. 
What is your opinion?